.github/workflows/static-code-analysis.yml

```
@@ -13,13 +13,17 @@ jobs:
                 auth_header="$(git config --local --get http.https://github.com/.extraheader)"
                 git submodule sync --recursive
                 git -c "http.extraheader=$auth_header" -c protocol.version=2 submodule update --init --force --recursive --depth=1
-            - name: Remove composer.json
-              shell: bash
-              run: rm composer.json composer.lock
-            - name: Psalm
-              uses: docker://jakzal/phpqa:php7.4-alpine
+            - name: Set up php7.4
+              uses: shivammathur/setup-php@master
               with:
-                args: psalm --monochrome --no-progress --output-format=text --update-baseline || ( git diff && exit 1 )
+                php-version: 7.4
+                coverage: none
+            - name: Composer install
+              run: composer i
+            - name: Psalm
+              run: composer run psalm -- --monochrome --no-progress --output-format=text --update-baseline || ( git diff -- . ':!lib/composer' && exit 1 )
+            - name: Check diff
+              run: git diff -- . ':!lib/composer' && exit 1
```

```suggestion
              run: git diff -- . ':!lib/composer'
```

We should not error on fixed issues. There is a daily GitHub workflow that commits this.